spock-specs/src/test/groovy/org/spockframework/smoke/parameterization/UprolledFeatureMethods.groovy

```diff
@@ -0,0 +1,257 @@
+    when:
+    runner.runSpecBody """
+$annotation
+def foo() {
+  expect: true
+
+  where:
+  x << []
+}
+    """
+
+    then:
+    SpockExecutionException e = thrown()
+
+    where:
+    annotation << ["", "@Rollup"]
+  }
+
+  def "can roll up a whole class at once"() {
+    when:
+    def result = runner.runWithImports("""
+@Rollup
+class Foo extends Specification {
+  def "#actor.details.name"() {
+    expect: true
+
+    where:
+    actor = new Actor()
+  }
+
+  def "not data-driven"() {
+    expect: true
+  }
+
+  def "#actor.details.age"() {
+    expect: true
+
+    where:
+    actor = new Actor()
+  }
+}
+    """)
+
+    then:
+    result.testEvents().started().list().testDescriptor.displayName == ["#actor.details.name",
+                                                                        "not data-driven",
+                                                                        "#actor.details.age"]
+  }
+
+  def "method-level uproll annotation wins over class-level unroll annotation"() {
+    when:
+    def result = runner.runWithImports("""
+@Unroll("#actor.details.name")
+class Foo extends Specification {
+  @Rollup
+  def method() {
+    expect: true
+
+    where:
+    actor = new Actor()
+  }
+}
+    """)
+
+    then:
+    result.testEvents().started().list().testDescriptor.displayName == ["method"]
+  }
+
+  def "unroll and uproll cannot be combined on the same spec"() {
+    when:
+    runner.runWithImports("""
+@Unroll("#actor.details.name")
+@Rollup
+class Foo extends Specification {
+  def method() {
+    expect: true
+
+    where:
+    actor = new Actor()
+  }
+}
+    """)
+
+    then:
+    thrown(InvalidSpecException)
+  }
+
+  def "unroll and uproll cannot be combined on the same feature"() {
+    when:
+    runner.runSpecBody("""
+@Unroll("#actor.details.name")
+@Rollup
+def method() {
+  expect: true
+
+  where:
+  actor = new Actor()
+}
+    """)
+
+    then:
+    thrown(InvalidSpecException)
+  }
+
+  def "method-level uproll annotation wins over configuration script"() {
+    given:
+    runner.configurationScript {
+      unroll {
+        unrollByDefault true
+      }
+    }
+
+    when:
+    def result = runner.runSpecBody("""
+@Rollup
+def method() {
+  expect: true
+
+  where:
+  actor = new Actor()
+}
+    """)
+
+    then:
+    result.testEvents().started().list().testDescriptor.displayName == ["method"]
+  }
+
+  def "class-level uproll annotation wins over configuration script"() {
+    given:
+    runner.configurationScript {
+      unroll {
+        unrollByDefault true
+      }
+    }
+
+    when:
+    def result = runner.runWithImports("""
+@Rollup
+class Foo extends Specification {
+  def method() {
+    expect: true
+
+    where:
+    actor = new Actor()
+  }
+}
+    """)
+
+    then:
+    result.testEvents().started().list().testDescriptor.displayName == ["method"]
+  }
+
+  def "rollup annotations are not inheritable"() {
```

Comment:
Fehlt nicht noch einer für `@Unroll` oder ist das schon wo anders?

Reply:
Der ist in `UnrolledFeatureMethods#unroll annotations are not inheritable`